ArchiSteamFarm/Bot.cs

```diff
@@ -3912,6 +3920,163 @@ internal sealed class Bot : IDisposable {
+			}
+
+			// Not sure if I need to keep it, guess it's better.
+			if (!LootingAllowed) {
+				return FormatBotResponse(Strings.BotLootingTemporarilyDisabled);
+			}
+			Bot botTo;
+			if(!Bots.TryGetValue(botNameTo, out botTo)) {
+				return IsOwner(steamID) ? FormatStaticResponse(string.Format(Strings.BotNotFound, botNameTo)) : null;
+			}
+			
+			ulong targetSteamMasterID = botTo.SteamID;
+
+			if (targetSteamMasterID==0) {
+				return FormatBotResponse(Strings.BotNotConnected);
+			}
+
+			if (targetSteamMasterID == SteamID) {
+				return FormatBotResponse(Strings.BotLootingYourself);
+			}
+
+			if (!LootingSemaphore.Wait(0)) {
+				return FormatBotResponse(Strings.BotLootingFailed);
+			}
+
+			try {
+				HashSet<Steam.Item.EType> transferTypes = new HashSet<Steam.Item.EType>();
+				string[] modes = mode.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+				foreach (string singleMode in modes) {
+					switch (singleMode.ToUpper()) {
+						case "C": // Not sure about shortcuts.
+						case "CARD":
+							transferTypes.Add(Steam.Item.EType.TradingCard);
+							break;
+						case "F":
+						case "FOIL":
+							transferTypes.Add(Steam.Item.EType.FoilTradingCard);
+							break;
+						case "B":
+						case "BOOSTER":
+							transferTypes.Add(Steam.Item.EType.BoosterPack);
+							break;
+						case "E":
+						case "EMOTICON":
+							transferTypes.Add(Steam.Item.EType.Emoticon);
+							break;
+						case "BA":
+						case "BACKGROUND":
+							transferTypes.Add(Steam.Item.EType.ProfileBackground);
+							break;
+						case "U":
+						case "UNKNOWN":
+							transferTypes.Add(Steam.Item.EType.Unknown);
+							break;
+						case "G":
+						case "GEMS":
+							transferTypes.Add(Steam.Item.EType.SteamGems);
+							break;
+						case "EV":
+						case "EVERYTHING":
+							// Needs to be kept up to date, or is there an easy way for all types?
+							transferTypes.Add(Steam.Item.EType.TradingCard);
+							transferTypes.Add(Steam.Item.EType.FoilTradingCard);
+							transferTypes.Add(Steam.Item.EType.BoosterPack);
+							transferTypes.Add(Steam.Item.EType.Emoticon);
+							transferTypes.Add(Steam.Item.EType.ProfileBackground);
+							transferTypes.Add(Steam.Item.EType.Unknown);
+							transferTypes.Add(Steam.Item.EType.SteamGems);
+							break;
+						default:
+							// adjust error message
+							return "Unknown mode " + singleMode + "!";
+					}
+				}
+
+				HashSet<Steam.Item> inventory = await ArchiWebHandler.GetMySteamInventory(true, transferTypes).ConfigureAwait(false);
+				if ((inventory == null) || (inventory.Count == 0)) {
+					return FormatBotResponse(string.Format(Strings.ErrorIsEmpty, nameof(inventory)));
+				}
+
+				if (!await ArchiWebHandler.MarkSentTrades().ConfigureAwait(false)) {
+					return FormatBotResponse(Strings.BotLootingFailed);
+				}
+
+				if (SteamFriends.GetFriendRelationship(targetSteamMasterID)== EFriendRelationship.Friend) {
```

Comment:
This can be simplified, you can set token as `null` and change it from null only inside `SteamFriends.GetFriendRelationship(targetSteamMasterID) != EFriendRelationship.Friend` then pass that token (modified or not) to `ArchiWebHandler.SendTradeOffer()` always. If token is null or empty, it's not being included. This is why that field is optional and set to null by default. This is misc though.